### DISCUSSION TITLE
eclipse-updatesite in maven local repo and dependency resolution

### DISCUSSION THREAD

**Discussion by umairsair** [eclipse-tycho/tycho#452] [Q&A] [answered]
Hi,

I have an eclipse-updatesite installed in my local maven repository. My eclipse-feature build is failing because of missing IU requirement but that IU exists in that updatesite.

Does tycho consider the updatesite installed in local repo for dependencies and target platform resolution?

**Comment by laeubi** [comment, marked as answer]
You need to reference the update-site with its maven coordinates in your pom fro it to be considered.

```
<repository>
   <id>my-maven-deployed-site</id>
   <url>mvn:[grouId]:[artifactId]:[version]:[extension]:[qualifier]</url>
   <layout>p2</layout>
</repository>
```

**Reply by umairsair** [reply]
I got following error. Did I provide it in correct way?

```org.apache.maven.InternalErrorException: Internal error: java.lang.RuntimeException: Invalid repository URL: mvn:my.example.updatesite:myupdatesite:12.4.0:zip:202112141841```

**Reply by laeubi** [reply]
The best would be I think to copy the regular artifacts dependency the number at the end looks a bit uncommon.

**Reply by umairsair** [reply]
The problem seems to be because of protocol.

```
Caused by: java.net.MalformedURLException: unknown protocol: mvn
        at java.net.URL.<init>(URL.java:600)
        at java.net.URL.<init>(URL.java:490)
        at java.net.URL.<init>(URL.java:439)
```

**Reply by laeubi** [reply]
Can you share an example? Are you using the latest Tycho version (2.5.0)?

**Reply by umairsair** [reply]
no. I am on older version. Let me try 2.5.0.

**Reply by umairsair** [reply]
I am using Java 8 :(

which tycho version should I try?

**Reply by laeubi** [reply]
Sorry this feature is only available for Tycho 2.x. This requires you to run with Java 11, but you can still build/compile with java 8.

**Reply by umairsair** [reply]
It worked with Java 11 and Tycho 2.2.

Thanks.
